chalice/deploy/deployer.py

```diff
@@ -280,7 +308,48 @@ def deploy(self, config, existing_resources, stage_name):
                 config, function_name, stage_name)
             deployed_values['api_handler_name'] = function_name
         deployed_values['api_handler_arn'] = function_arn
-        return deployed_values
+
+    def _deploy_auth_handlers(self, config, existing_resources, stage_name,
```

Comment:
It will be semi-orphaned in the sense that we won't delete it on deploy, but it will still stay in `deployed.json` so will get removed as part of `chalice delete`.

I agree it does make sense if you delete your auth handler code we should remove the lambda function.  I'll look into how much work that is.